### PR TITLE
Core: Add previewHead and previewBody to StorybookConfig interface

### PR DESCRIPTION
**Pull request by mandarini**
Issue: 

`previewHead` does not exist in the StorybookConfig interface. This results in type errors when using TypeScript configuration (`main.ts`). See screenshot:

Screenshot 2022-08-29 at 8 03 31 PM

## What I did

I added `previewHead` and `previewBody` in the StorybookConfig interface.

This was raised [in this issue](https://github.com/nrwl/nx/issues/2838#issuecomment-1230566486) on the Nx side.

### CODE REVIEW

File: code/lib/core-common/src/types.ts [hunk 1]
```diff
@@ -428,6 +428,15 @@ export interface StorybookConfig {
    * Docs related features in index generation
    */
   docs?: DocsOptions;
+
+  /**
+   * Programmatically modify the preview head/body HTML.
+   * The previewHead and previewBody functions accept a string,
+   * which is the existing head/body, and return a modified string.
+   */
+  previewHead: (head: any) => string;
```
[the remark this answers is not in the document]
Comment (mandarini) [reply code/lib/core-common/src/types.ts:437]: Oh yeah I guess this makes sense, oops!

[the remark this answers is not in the document]
Comment (mandarini) [reply code/lib/core-common/src/types.ts]: fixed it

[the remark this answers is not in the document]
Comment (yannbf) [reply code/lib/core-common/src/types.ts]: I guess it matters to define whether the head parameter should be optional or not. Not sure about it!

File: code/lib/core-common/src/types.ts [hunk 2]
```diff
@@ -428,6 +428,15 @@ export interface StorybookConfig {
    * Docs related features in index generation
    */
   docs?: DocsOptions;
+
+  /**
+   * Programmatically modify the preview head/body HTML.
+   * The previewHead and previewBody functions accept a string,
+   * which is the existing head/body, and return a modified string.
+   */
+  previewHead?: (head: string) => string;
```
Comment (shilman) [inline code/lib/core-common/src/types.ts:437]:
```suggestion
  previewHead?: (head: string, options: Options) => string;
```

File: code/lib/core-common/src/types.ts [hunk 3]
```diff
@@ -428,6 +428,15 @@ export interface StorybookConfig {
...
+
+  previewBody?: (body: string) => string;
```
Comment (shilman) [inline code/lib/core-common/src/types.ts:439]:
```suggestion
  previewBody?: (body: string, options: Options) => string;
```

**Review by shilman — APPROVED** [review]
Thanks @mandarini !!!! Looks great with a small change

File: code/lib/core-common/src/types.ts (hunk 2 above)
Comment (mandarini) [reply code/lib/core-common/src/types.ts]: Oh thanks! :) Should options be optional, though?

**Pull request merged**